spec.md

```diff
@@ -535,9 +540,55 @@ message PluginCapability {
     Type type = 1;
   }
 
+  message VolumeExpansion {
+    enum Type {
+      UNKNOWN = 0;
+
+      // ONLINE indicates that volumes may be expanded when published to
+      // a node. When a Plugin implements this capability it MUST
+      // implement either the EXPAND_VOLUME controller capability or the
+      // EXPAND_VOLUME node capability or both. When a plugin supports
+      // ONLINE volume expansion and also has the EXPAND_VOLUME
+      // controller capability then the plugin MUST support expansion of
+      // volumes currently published and available on a node. When a
+      // plugin supports ONLINE volume expansion and also has the
+      // EXPAND_VOLUME node capability then the plugin MAY support
+      // expansion of node-published volume via NodeExpandVolume.
+      //
+      // Example 1: Given a shared filesystem volume (e.g. GlusterFs),
+      //   the Plugin may set the ONLINE volume expansion capability and
+      //   implement ControllerExpandVolume but not NodeExpandVolume.
+      //
+      // Example 2: Given a block storage volume type (e.g. EBS), the
+      //   Plugin may set the ONLINE volume expansion capability and
+      //   implement both ControllerExpandVolume and NodeExpandVolume.
+      //
+      // Example 3: Given a Plugin that supports volume expansion only
+      //   on upon a node, the Plugin may set the ONLINE volume
+      //   expansion capability and implement NodeExpandVolume but not
+      //   ControllerExpandVolume.
+      ONLINE = 1;
+
+      // OFFLINE indicates that volumes currently published and
+      // available on a node MAY NOT be expanded via
+      // ControllerExpandVolume. When a plugin support
+      // OFFLINE volume expansion it MUST implement either the
+      // EXPAND_VOLUME controller capability or the EXPAND_VOLUME
```

Comment:
suggest: "... it MUST implement either the EXPAND_VOLUME controller capability, or both the EXPAND_VOLUME controller capability and the EXPAND_VOLUME node capability."

... because it doesn't make sense to combine OFFLINE with *only* the EXPAND_VOLUME node capability. OFFLINE only applies to the controller capability, so if someone wants to implement *only* the node capability then I'd rather the spec provide a single way for them to indicate that: ONLINE + node capability.

Reply:
This is exactly what I want to point out. Combining `OFFLINE` with *only* the `EXPAND_VOLUME` node capability is just contradictory.

Reply:
ack. I will update the wording to reflect this.I think ONLINE + node capability already covers the bit about if a plugin wants to support *only* node expansion capability.

Reply:
updated.